CHANGES.md

```diff
@@ -1,6 +1,417 @@
+* enabled building with GPU support for host compilers other than `clang`
+
+Bug Fixes for Libraries
+-----------------------
+* fixed a bug with `bigint.removeFactor()` when `fac` is `0`
+
+Third-Party Software Changes
+----------------------------
+* updated the bundled copy of LLVM and clang to version 13
+* updated the bundled copy of `libfabric` to version 1.13.2
+* updated the version of Sphinx used for `chpldoc` to 4.3.2
+* updated the Sphinx Chapel domain version used for `chpldoc` to 0.0.21
+* updated the breathe version used for `chpldoc` to 4.31.0
+
+Developer-oriented changes: Process
+-----------------------------------
+* trailing whitespace is now flagged in CI using GitHub actions
+
+Developer-oriented changes: Documentation
+-----------------------------------------
+* removed references to outdated files from the `chplspell` documentation
+* fixed the path to 'ChapelIO' in 'How To Generate Warnings And Error Messages'
+
+Developer-oriented changes: Naming Changes
+------------------------------------------
+* coined the name 'dyno' to refer to aspects of the compiler rewrite effort
+
+Developer-oriented changes: Module changes
+------------------------------------------
+* simplified the barrier used to initialize locales
+* updated the 'Socket' module's operators to use the `operator` keyword
+* added `iostyleInternal` type for use when `iostyle` is required by developers
+
+Developer-oriented changes: Makefile / Build-time changes
+---------------------------------------------------------
+* adjusted the unique build paths for `hwloc` and `re2`
+* stopped inappropriately building the bundled GMP when `CHPL_GMP=system`
+* improved ordering of `-I`/`-L` arguments for bundled vs. system versions
+* non-debug builds of `chpl` now use `-DNDEBUG` to disable asserts in C++ code
+* removed the `make third-party` target
+* a 'dyno'-specific C++ linter now checks some class methods
+* stopped relying on `distutils`, which is deprecated in Python 3.10
+
+Developer-oriented changes: Compiler Flags
+------------------------------------------
+* added support for a `-j`/`--parallel-make` flag for the C back-end
+* renamed the `--compiler-library-parser` flag to `--dyno`
+* added `--dyno-debug-trace`/`--dyno-break-on-hash` for debugging 'dyno'
+
+Developer-oriented changes: Compiler improvements/changes
+---------------------------------------------------------
+* updated `--incremental` to compile each module's `.c` file independently
+* enabled support for `make -j` using the C back-end via `-j` / `--incremental`
+* added internal modules to the output of `--devel --print-callstack-on-error`
+* stopped relying on `distutils`, which is deprecated in Python 3.10
+
+Developer-oriented changes: 'dyno' Compiler improvements/changes
+----------------------------------------------------------------
+* the 'dyno' front-end can now parse all primers and modules
+* improved `--dyno` to the point that ~13,222/13,697 tests now pass
```

Comment:
Seems a bit unusual to me to include a snapshot of the number of passing tests, but I couldn't articulate why.

Reply:
I felt similarly, but also feel proud of the numbers and think it indicates progress that an invested developer might care about.  Could express as a %age which  maybe feels slightly less weird / arbitrary.

Reply:
Fine with me either way. %age feels cleaner, but the number of tests is impressive.

Reply:
After typing the above, I computed the %age and switched to it.